<!doctype html>
<html lang="ja">

<head>
	<title>ConnectorException 仕様書</title>
	<meta charset="UTF-8">
	<style type="text/css">
		h1 .spec-path { font-size: 70%; }
		h2 .spec-path { font-size: 70%; }
		.spec-table { width: 100%; border: 1px solid #000000; border-collapse: collapse; margin-bottom: 20px; }
		.spec-table td { border: 1px solid #000000; text-align: left; }
		.spec-table th { border: 1px solid #000000; text-align: left; }
		.spec-table .lcol { width: 100px; background-color: #eeeeee; }
		.spec-table .rcol { }
		.spec-dl { }
		.spec-dl dt { font-weight: bold; display: list-item; list-style-position: outside; margin-left: 20px; list-style-type: disc; }
		.spec-dl dd { margin-bottom: 14px; line-height: 160%; }
	</style>
</head>

<body>



<div style="text-align: center; width: 100%; border: 1px solid #000000; background-color: #eeeeff;">
VCSSL/Vnano プラグイン インターフェース 仕様書
</div>


<h1>ConnectorException<br />クラス仕様書<br /><span class="spec-path">(org.vcssl.connect.ConnectorException)</span></h1>

<hr />



<h2>概要</h2>

<p>
プラグインにおいて、実行時に生じた問題を、処理系側に通知するための検査例外です。
</p>

<p>
この例外は、例えばファイルへのアクセス失敗時など、
問題が「プラグインやスクリプトエンジンの実装が正しくても生じ得る」場合にスローしてください。
</p>

<p>
両者の実装が正しければ生じない（つまりバグの可能性が高い）問題に対しては、この例外の代わりに、非検査例外である 
<a href="ConnectorFatalException_SPEC_JAPANESE.html">ConnectorFatalException</a> をスローしてください。
</p>



<h2>ステータス</h2>

<p>
このクラスの現在のステータスは "EXTENSION ONLY"（拡張のみ）です。
</p>

<p>
既存のコンストラクタ/メソッド/フィールドは原則として削除されませんが、将来的に新規追加される事はあり得ます。
<br />
また、どうしてもやむを得ない事情があった場合のみ、非推奨化の段階を経た上で、各要素の削除が検討される事もあり得ます。
</p>

<p>
このクラスは、現在は <a href="https://www.vcssl.org/ja-jp/vnano/">Vnano</a> のスクリプトエンジンでサポートされ、同エンジンを搭載したアプリケーションのプラグイン開発において利用できます。
</p>


<h2>ライセンス</h2>

<p>
このクラスは、実質的な著作権フリー/パブリックドメインである CC0 に基づいて公開されています。
</p>


<h2>コンストラクタ/フィールド/メソッド等の一覧</h2>

<dl class="spec-dl">

	<dt><a href="#ConnectorException1">ConnectorException()</a></dt>
		<dd>
			エラーメッセージを持たない ConnectorException を生成します。
		</dd>

	<dt><a href="#ConnectorException2">ConnectorException(String errorMessage)</a></dt>
		<dd>
			指定されたエラーメッセージを持つ ConnectorException を生成します。
		</dd>

	<dt><a href="#ConnectorFatalException3">ConnectorException(Throwable wrappedThrowable)</a></dt>
		<dd>
			原因情報を保持する、指定された Throwable をラップする ConnectorException を生成します。
		</dd>

	<dt><a href="#ConnectorFatalException4">ConnectorException(String errorMessage, Throwable wrappedThrowable)</a></dt>
		<dd>
			指定されたエラーメッセージを持ち、指定された Throwable（原因情報を保持）をラップする ConnectorException を生成します。
		</dd>

</dl>



<hr />



<!-- ==================================================================================================== -->

<h2>コンストラクタ</h2>

<table id="ConnectorException1" class="spec-table">
	<tr>
		<th class="lcol">名前</th>
		<th class="rcol">ConnectorException</th>
	</tr>
	<tr>
		<td class="lcol">宣言形式</td>
		<td class="rcol">ConnectorException()</td>
	</tr>
	<tr>
		<td class="lcol">機能</td>
		<td class="rcol">エラーメッセージを持たない ConnectorException を生成します。</td>
	</tr>
	<tr>
		<td class="lcol">引数</td>
		<td class="rcol">なし</td>
	</tr>
	<tr>
		<td class="lcol">戻り値</td>
		<td class="rcol">生成された ConnectorException インスタンス。</td>
	</tr>
</table>


<table id="ConnectorException2" class="spec-table">
	<tr>
		<th class="lcol">名前</th>
		<th class="rcol">ConnectorException</th>
	</tr>
	<tr>
		<td class="lcol">宣言形式</td>
		<td class="rcol">ConnectorException(String errorMessage)</td>
	</tr>
	<tr>
		<td class="lcol">機能</td>
		<td class="rcol">指定されたエラーメッセージを持つ ConnectorException を生成します。</td>
	</tr>
	<tr>
		<td class="lcol">引数</td>
		<td class="rcol">errorMessage: エラーメッセージ。</td>
	</tr>
	<tr>
		<td class="lcol">戻り値</td>
		<td class="rcol">生成された ConnectorException インスタンス。</td>
	</tr>
</table>


<table id="ConnectorException3" class="spec-table">
	<tr>
		<th class="lcol">名前</th>
		<th class="rcol">ConnectorException</th>
	</tr>
	<tr>
		<td class="lcol">宣言形式</td>
		<td class="rcol">ConnectorException(Throwable wrappedThrowable)</td>
	</tr>
	<tr>
		<td class="lcol">機能</td>
		<td class="rcol">原因情報を保持する、指定された Throwable をラップする ConnectorException を生成します。</td>
	</tr>
	<tr>
		<td class="lcol">引数</td>
		<td class="rcol">wrappedThrowable: 原因情報を保持する、ラップ対象の Throwable。</td>
	</tr>
	<tr>
		<td class="lcol">戻り値</td>
		<td class="rcol">生成された ConnectorException インスタンス。</td>
	</tr>
</table>


<table id="ConnectorException4" class="spec-table">
	<tr>
		<th class="lcol">名前</th>
		<th class="rcol">ConnectorException</th>
	</tr>
	<tr>
		<td class="lcol">宣言形式</td>
		<td class="rcol">ConnectorException(String errorMessage, Throwable wrappedThrowable)</td>
	</tr>
	<tr>
		<td class="lcol">機能</td>
		<td class="rcol">指定されたエラーメッセージを持ち、指定された Throwable（原因情報を保持）をラップする ConnectorException を生成します。</td>
	</tr>
	<tr>
		<td class="lcol">引数</td>
		<td class="rcol">
			errorMessage: エラーメッセージ。
			<br />
			wrappedThrowable: 原因情報を保持する、ラップ対象の Throwable。
		</td>
	</tr>
	<tr>
		<td class="lcol">戻り値</td>
		<td class="rcol">生成された ConnectorException インスタンス。</td>
	</tr>
</table>



<!-- ==================================================================================================== -->

<h2>フィールド</h2>

<p>
- なし -
</p>





<!-- ==================================================================================================== -->

<h2>メソッド</h2>


<p>
- なし -
</p>









</body>
</html>
